engine/src/main/java/org/camunda/bpm/engine/impl/pvm/runtime/operation/PvmAtomicOperationActivityStart.java

```diff
@@ -49,4 +53,20 @@ public String getCanonicalName() {
   public boolean shouldHandleFailureAsBpmnError() {
     return true;
   }
+
+  @Override
+  protected void setDelayedPayloadToNewScope(PvmExecutionImpl execution, CoreModelElement scope) {
+    String activityType = (String) scope.getProperty(BpmnProperties.TYPE.getName());
+    if (ActivityTypes.START_EVENT_MESSAGE.equals(activityType) // Event subprocess message start event
+        || ActivityTypes.BOUNDARY_MESSAGE.equals(activityType)) {
+      PvmExecutionImpl processInstance = execution.getProcessInstance();
+      ActivityNewScopeVariablesTuple tuple = processInstance.getPayloadForTriggeredScope();
+      if (tuple != null && tuple.getActivityId().equals(scope.getId())) {
```

Comment:
❓What is the idea of `tuple.getActivityId().equals(scope.getId())`? In which situations would the `activityId`s not match? I initially thought we needed the `activityId` to figure out the right scope. But how it's built now, I wonder why we need it. Even when correlating multiple messages on different flow nodes in parallel with `setVariablesToTriggeredScope` for each correlation, we have dedicated execution objects, so it shouldn't happen that this method is called and the `activityId`s don't match, or am I missing something? 🤔

Reply:
Good point, I will check.